rust/azure_iot_operations_protocol/src/telemetry/sender.rs

```diff
@@ -732,4 +735,27 @@ mod tests {
 
         assert!(message_builder_result.is_err());
     }
+
+    #[test]
+    fn test_send_retain_true() {
+        let mut mock_telemetry_payload = MockPayload::new();
+        mock_telemetry_payload
+            .expect_serialize()
+            .returning(|| {
+                Ok(SerializedPayload {
+                    payload: String::new().into(),
+                    content_type: "application/json".to_string(),
+                    format_indicator: FormatIndicator::Utf8EncodedCharacterData,
+                })
+            })
+            .times(1);
+
+        let message_builder_result = MessageBuilder::default()
+            .payload(mock_telemetry_payload)
+            .unwrap()
+            .retain(true)
+            .build();
+
+        assert!(message_builder_result.unwrap().retain);
```

Comment:
This isn't really testing anything. It's testing that the builder sets the `retain` value as far as I can tell?
That's what a builder does, and it's derived code at that.

I'm not sure a test here is needed really, and if there is one required, there almost certainly shouldn't be one exclusively for testing `retain` - there should be a broad test of the Message struct

Reply:
now testing defaults